open-craft/frontend-lib-content-components
# feat: olx parser for problem

## Description

This is PR includes two parsers:

- [OLX in parser](https://github.com/open-craft/frontend-lib-content-components/pull/4/commits/9a28e02285b5444a29bcbd51fa1c9efdd52f4450) which converts the OLX to  subsequent React State
- [React state to OLX parser](https://github.com/open-craft/frontend-lib-content-components/pull/4/commits/9aef656fad966fad1dfc9a2ab435056ecd968841) this convert the react state to OLX

The code is coverting the all the problem information to and fro the edx-platform.

Useful information to include:

None

## Supporting information

Private ref: [BB-6706](https://tasks.opencraft.com/browse/BB-6706)

## Testing instructions

There are few test cases that are written which can be run by:

- `npm test -- src/editors/containers/ProblemEditor/data/ReactStateOLXParser.test.js --collectCoverage=false`
- `npm test -- src/editors/containers/ProblemEditor/data/OLXParser.test.js --collectCoverage=false`

The conversion can be looked into [mock olx data](https://github.com/open-craft/frontend-lib-content-components/pull/4/files#diff-67af2bfbd78e96e31f5409dea0db43b73dd140c40c0af762d2b1ccad77549d5e)

The decision to covert lable and description into bold and em tag was taken since lable and descriptiopn weren't rendering the way they should.

We can get the devstack working and edit any of the basic problem and the preview block of xblock should reflect it.

## Deadline

None

## Other information

None
